oidc/provider_test.go

```diff
@@ -742,6 +744,7 @@ func TestProviderConfigSyncerRun(t *testing.T) {
 		if tt.count != from.getCount {
 			t.Fatalf("test %d: want: %v, got: %v", i, tt.count, from.getCount)
 		}
+		cancel()
```

Comment:
No really necessary with the defer, right?

Reply:
I was on the fence as to what's best.
1) If we only use the defer in the for loop, then a very large amount of test cases may result in excessive resource consumption
2) If we don't defer and only cancel here then in test conditions a panic could result in a routine not canceling
3) I could put it in an anonymous function and just use defer, but the line change would be more dramatic

I provided both 1 & 2 in an effort to be both defensive and minimal in the change.  But really I don't have a strong preference, it's an optimization in tests that I don't think will have much impact.  So I will defer to whoever works in this code most often.